src/Chapter4.hs

```diff
@@ -600,7 +624,8 @@ Implement the 'Monad' instance for our 'Secret' type.
 -}
 instance Monad (Secret e) where
     (>>=) :: Secret e a -> (a -> Secret e b) -> Secret e b
-    (>>=) = error "bind Secret: Not implemented!"
+    (>>=) (Trap e) _ = Trap e
+    (>>=) (Reward a) f = (f a)
```

Comment:
This is nice!
In Haskell you can write operators as operators at the left side of `=` as well:

```suggestion
    Trap e >>= _ = Trap e
    Reward a >>= f = f a
```